DEMO.md

```diff
@@ -202,6 +202,42 @@ that they confuse CMake:
 Likely, this breaks CMake's platform detection somehow, but there could be
 something else going on.
 
+
+#### Build swift compiler
+
+In order to build swift compiler from `main` branch which is not CAS aware, do:
+
+```
+% CCC_OVERRIDE_OPTIONS="+-fdepscan +-fdepscan-share-parent=ninja +-fdepscan-share-stop=cmake" \
+  ./utils/build-script --host-cc $CLANG --host-cxx ${CLANG}++ $OTHER_FLAGS
+```
+
+Or if you use the pre-built cas-aware toolchain, you can do:
+
+```
+# only need to run the first command once, maybe include that in the future toolchain?
+% (cd $TOOLCHAIN && ln -s clang cc)
+% CCC_OVERRIDE_OPTIONS="+-fdepscan +-fdepscan-share-parent=ninja +-fdepscan-share-stop=cmake"
+	TOOLCHAINS="Local Swift Development Snapshot" ./util/build-script $OTHER_FLAGS
+```
+
+Or if you are on the cas-aware swift branch, checkout from `experimental/cas/rebranch` branch via:
+
+```
+% ./util/update-checkout --scheme experimental/cas/rebranch
+```
+
+You can build with a CAS aware toolchain:
+
+```
+% TOOLCHAINS="Local Swift Development Snapshot" ./util/build-script --cas $OTHER_FLAGS
+```
+
+FIXME:
+
+* When you are normal swift branch, there is no easy way to pass c flags to each sub component (?) so we use `CCC_OVERRIDE_OPTIONS` to overwrite it globally. The downside is we can't apply path fixup or enable other caching features (like TableGen). It can also turn on `-fdepscan` in places that we never heavily tested (like within cmake configuration, swift package manager, etc.).
```

Comment:
I think what's important to explain is that build script doesn't provide access for adding options to `-DCMAKE_{C,CXX}_FLAGS='...'`. Then the FIXME is for adding options to build script that support customizing that.

Reply:
(LGTM once you cover that!)

Reply:
I will just get rid of FIXME because that is mostly handled on my branch already.